src/core/channelz/channelz_registry.h

```diff
@@ -146,52 +139,30 @@ class ChannelzRegistry final {
     BaseNode* head = nullptr;
     BaseNode* tail = nullptr;
     size_t count = 0;
-    bool Holds(BaseNode* node) const {
-      BaseNode* n = head;
-      while (n != nullptr) {
-        if (n == node) return true;
-        n = n->next_;
-      }
-      return false;
-    }
-    void AddToHead(BaseNode* node) {
-      DCHECK(!Holds(node));
-      ++count;
-      if (head != nullptr) head->prev_ = node;
-      node->next_ = head;
-      node->prev_ = nullptr;
-      head = node;
-      if (tail == nullptr) tail = node;
-      DCHECK(Holds(node));
-    }
-    void Remove(BaseNode* node) {
-      DCHECK(Holds(node));
-      DCHECK_GT(count, 0);
-      --count;
-      if (node->prev_ == nullptr) {
-        head = node->next_;
-        if (head == nullptr) {
-          DCHECK_EQ(count, 0);
-          tail = nullptr;
-          DCHECK(!Holds(node));
-          return;
-        }
-      } else {
-        node->prev_->next_ = node->next_;
-      }
-      if (node->next_ == nullptr) {
-        tail = node->prev_;
-      } else {
-        node->next_->prev_ = node->prev_;
-      }
-      DCHECK(!Holds(node));
-    }
+    bool Holds(BaseNode* node) const;
+    void AddToHead(BaseNode* node);
+    void Remove(BaseNode* node);
   };
+  // Nodes traverse through up to four lists, depending on
+  // whether they have a uuid (this is becoming numbered),
+  // and whether they have been orphaned or not.
+  // The lists help us find un-numbered nodes when needed for
+  // queries, and the oldest orphaned node when needed for
+  // garbage collection.
+  // Nodes are organized into shards based on their pointer
+  // address. A shard tracks the four lists of nodes
+  // independently - we strive to have no cross-talk between
```

Comment:
Do we actually have a way of ensuring that if a node references another node, that other node is in the same shard?  Or is this more of a "we'd ideally like this but can't actually do it" kind of statement?  If the latter, please clarify the wording.

Reply:
We don't... I thought strive covered this... it certainly doesn't say we guarantee that.